backend/src/main/java/gov/cdc/usds/simplereport/service/ResultsUploaderDeviceService.java

```diff
@@ -0,0 +1,100 @@
+    }
+    return value;
+  }
+
+  private static String addTrailingAsterisk(String value) {
+    if (value != null && !value.isEmpty() && value.charAt(value.length() - 1) != '*') {
+      return value + "*";
+    }
+    return value;
+  }
+
+  public DeviceType getDeviceFromCache(String model, String testPerformedCode) {
+    // often our devices have a trailing asterisk in the model name, which users often leave out and
+    // generate support burden. In those cases, try both and return the non-null value.
+
+    Map<String, DeviceType> referenceList =
+        resultsUploaderCachingService.getModelAndTestPerformedCodeToDeviceMap();
+    String keyToCheckWithAsterisk =
+        ResultsUploaderCachingService.getKey(addTrailingAsterisk(model), testPerformedCode);
+
+    String keyToCheckWithoutAsterisk =
+        ResultsUploaderCachingService.getKey(removeTrailingAsterisk(model), testPerformedCode);
+
+    return Objects.requireNonNullElse(
+        referenceList.get(keyToCheckWithAsterisk), referenceList.get(keyToCheckWithoutAsterisk));
+  }
+
+  public boolean validateModelAndTestPerformedCombination(
+      String equipmentModelName, String testPerformedCode) {
+    Map<String, DeviceType> cachedResultsMap =
+        resultsUploaderCachingService.getModelAndTestPerformedCodeToDeviceMap();
+
+    boolean combinationKeyExistsWithoutAsterisk =
+        cachedResultsMap.containsKey(
+            ResultsUploaderCachingService.getKey(
+                removeTrailingAsterisk(equipmentModelName), testPerformedCode));
+
+    boolean combinationKeyExistsWithAsterisk =
+        cachedResultsMap.containsKey(
+            ResultsUploaderCachingService.getKey(equipmentModelName, testPerformedCode));
+
+    return equipmentModelName != null
+        && testPerformedCode != null
+        // the || here is for a similar reason to getResultDevice: often our devices have a trailing
+        // asterisk in the model name, which users often leave out and
+        // generate support burden. In those cases, try both and return the non-null value.
+        && (combinationKeyExistsWithoutAsterisk || combinationKeyExistsWithAsterisk);
+  }
+
+  public boolean validateResultsOnlyIncludeActiveDiseases(
+      String equipmentModelName, String testPerformedCode) {
+    if (equipmentModelName == null || testPerformedCode == null) {
+      return false;
+    }
+
+    DeviceType deviceTypeToCheck = getDeviceFromCache(equipmentModelName, testPerformedCode);
+
+    String supportedDiseaseNamesToCheck =
+        deviceTypeToCheck.getSupportedDiseaseTestPerformed().stream()
+            .filter(sdtp -> sdtp.getTestPerformedLoincCode().equals(testPerformedCode))
+            .map(DeviceTypeDisease::getSupportedDisease)
+            .map(SupportedDisease::getName)
+            .toList()
+            .get(0);
+
+    if (supportedDiseaseNamesToCheck.equals("HIV")) {
+      return featureFlagsConfig.isHivBulkUploadEnabled();
+    }
+    return true;
```

Comment:
intellij says that I can "simplify" this if / else to

```
return featureFlagsConfig.isHivBulkUploadEnabled()
              || !supportedDiseaseNamesToCheck.contains("HIV");
```
which is logically equivalent but is much less readable in my opinion. Leaving this in this form unless someone heavily disagrees